pandas/core/groupby/generic.py

```diff
@@ -1210,64 +1210,77 @@ def _wrap_applied_output(self, keys, values, not_indexed_same=False):
                 self._insert_inaxis_grouper_inplace(result)
                 return result
         else:
-            # this is to silence a DeprecationWarning
-            # TODO: Remove when default dtype of empty Series is object
-            kwargs = first_not_none._construct_axes_dict()
-            backup = create_series_with_explicit_dtype(dtype_if_empty=object, **kwargs)
-            values = [x if (x is not None) else backup for x in values]
-
-            all_indexed_same = all_indexes_same(x.index for x in values)
-
-            # GH3596
-            # provide a reduction (Frame -> Series) if groups are
-            # unique
-            if self.squeeze:
-                applied_index = self._selected_obj._get_axis(self.axis)
-                singular_series = len(values) == 1 and applied_index.nlevels == 1
-
-                # assign the name to this series
-                if singular_series:
-                    values[0].name = keys[0]
-
-                    # GH2893
-                    # we have series in the values array, we want to
-                    # produce a series:
-                    # if any of the sub-series are not indexed the same
-                    # OR we don't have a multi-index and we have only a
-                    # single values
-                    return self._concat_objects(
-                        keys, values, not_indexed_same=not_indexed_same
-                    )
+            # values are Series
+            return self._wrap_applied_output_series(
```

Comment:
isn't L1198 and L1207 also a series?

Reply:
L1198 `values` is a list of either ndarrays or Index; L1207 values are scalars.

Reply:
ok i c, can you make these clear in comments (followon ok).